homeassistant/components/eight_sleep/__init__.py

```diff
@@ -170,9 +199,25 @@ def __init__(
 
         mapped_name = NAME_MAP.get(sensor, sensor.replace("_", " ").title())
         if self._user_obj is not None:
-            mapped_name = f"{self._user_obj.side.title()} {mapped_name}"
-
-        self._attr_name = f"Eight {mapped_name}"
-        self._attr_unique_id = (
-            f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
-        )
+            assert self._user_obj.user_profile
+            name = f"{self._user_obj.user_profile['firstName']}'s {mapped_name}"
+            self._attr_name = name
+        else:
+            self._attr_name = f"Eight Sleep {mapped_name}"
+        unique_id = f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
```

Comment:
This variable is only used once. We can remove it by assigning the unique_id directly below.

Reply:
I split it into two lines because otherwise it would be three and I thought it was easier to read. Do you still think it's better to combine?

Reply:
Yes, in this case we don't use the variable for something else and we assign the value to the attribute on the line below. I think it's more succinct to assign directly. Otherwise I'm wondering where else we're using the local variable, when I'm reading this, and it's another variable to keep in your head.